moztelemetry/spark.py

```diff
@@ -162,39 +180,34 @@ def _get_ping_properties(ping, paths, only_median, with_processes,
     result = {}
 
     for property_name, path in paths:
-        cursor = ping
+        # Cursor will default to `PingCursor()` on missing keys
+        cursor = PingCursor(ping)
 
         if path[0] == "payload":
             path = path[1:]  # Translate v4 histogram queries to v2 ones
-            cursor = ping.get("payload", None)
+            cursor = cursor["payload"]
 
-            if cursor is None:
+            if not cursor:
                 return
 
         if path[0] == "histograms" or path[0] == "keyedHistograms":
             if path[0] == "keyedHistograms" and len(path) == 2:
                 # Include histograms for all available keys.
-                # These are returned as a subdict mapped from the property_name.
-                try:
-                    kh_keys = set(cursor["keyedHistograms"][path[1]].keys())
-                    if isinstance(cursor, dict):
-                        content_kh = cursor.get("processes", {}) \
-                                           .get("content", {}) \
-                                           .get("keyedHistograms", None)
-                        if content_kh is not None:
-                            kh_keys.update(content_kh[path[1]].keys())
-                        else:
-                            for payload in cursor.get("childPayloads", []):
-                                kh_keys.update(payload["keyedHistograms"][path[1]].keys())
-
-                        gpu_kh = cursor.get("processes", {}) \
-                                       .get("gpu", {}) \
-                                       .get("keyedHistograms", None)
-                        if gpu_kh is not None:
-                            kh_keys.update(gpu_kh[path[1]].keys())
-                except:
-                    result[property_name] = None
-                    continue
+                # These are returned as a subdict mapped from the
+                # property_name.
+                kh_keys = cursor["keyedHistograms"][path[1]].viewkeys()
+                if isinstance(cursor, dict):
+                    content_kh = cursor["processes"]["content"]["keyedHistograms"]
```

Comment:
It occurs to me that this will fail if `cursor["processes"]`, `cursor["processes"]["content"]`, etc. are present but not `dict`s. Do you know if that schema is mandated upstream?